src/codelens/bazel_build_code_lens_provider.ts

```diff
@@ -107,40 +150,62 @@ export class BazelBuildCodeLensProvider implements vscode.CodeLensProvider {
   ): vscode.CodeLens[] {
     const result: vscode.CodeLens[] = [];
 
+    interface LensCommand {
+      commandString: string;
+      name: string;
+    }
+
     for (const target of queryResult.target) {
       const location = new QueryLocation(target.rule.location);
       const targetName = target.rule.name;
       const ruleClass = target.rule.ruleClass;
-      let cmd: vscode.Command;
+      let targetShortName = getTargetShortName(
+        targetName,
+        bazelWorkspaceInfo,
+        location,
+      );
+
+      const commands: LensCommand[] = [];
+
+      // Only test targets support testing.
       if (ruleClass.endsWith("_test") || ruleClass === "test_suite") {
-        cmd = {
-          arguments: [
-            new CodeLensCommandAdapter(bazelWorkspaceInfo, [targetName]),
-          ],
-          command: "bazel.testTarget",
-          title: `Test ${targetName}`,
-          tooltip: `Test ${targetName}`,
-        };
-      } else if (ruleClass.endsWith("_binary")) {
-        cmd = {
-          arguments: [
-            new CodeLensCommandAdapter(bazelWorkspaceInfo, [targetName]),
-          ],
-          command: "bazel.runTarget",
-          title: `Run ${targetName}`,
-          tooltip: `Run ${targetName}`,
-        };
-      } else {
-        cmd = {
-          arguments: [
-            new CodeLensCommandAdapter(bazelWorkspaceInfo, [targetName]),
-          ],
-          command: "bazel.buildTarget",
-          title: `Build ${targetName}`,
-          tooltip: `Build ${targetName}`,
-        };
+        commands.push({
+          commandString: "bazel.testTarget",
+          name: "Test",
+        });
+      }
+
+      // Targets which are not libraries may support running.
+      //
+      // Without checking the Bazel rule's `executable` attribute we can't know
+      // for sure which targets can be run.
```

Comment:
This can be determined using a `cquery`, but this is quite a large change. Considering we used a heuristic before, I think this is fine.

```
bazel cquery '//:all' --output=starlark --starlark:expr "json.encode({'target': target.label.name, 'executable': providers(target)['FilesToRunProvider'].executable != None})"
```

Reply:
Yeah, I wasn't sure if this was worth the cost of the added query + latency. I'm happy to leave it as-is or make this change-- let me know which you'd prefer.

Reply:
I'm happy to leave it as-is, particularly because `cquery` has extra cost due to running the analysis phase.